proxy/proxy.go

```diff
@@ -57,56 +58,94 @@ type BaseURLResolver interface {
 // 	- logging errors and proxy request timing to stdout
 //
 // Note that this will panic if `resolver` is nil.
-func NewHandlerFunc(timeout time.Duration, resolver BaseURLResolver) http.HandlerFunc {
+func NewHandlerFunc(config types.FaaSConfig, resolver BaseURLResolver) http.HandlerFunc {
 	if resolver == nil {
 		panic("NewHandlerFunc: empty proxy handler resolver, cannot be nil")
 	}
 
-	proxyClient := http.Client{
+	proxyClient := NewProxyClientFromConfig(config)
+
+	return func(w http.ResponseWriter, r *http.Request) {
+		if r.Body != nil {
+			defer r.Body.Close()
+		}
+
+		switch r.Method {
+		case http.MethodPost,
+			http.MethodPut,
+			http.MethodPatch,
+			http.MethodDelete,
+			http.MethodGet:
+
+			proxyRequest(w, r, proxyClient, resolver)
+
+		default:
+			w.WriteHeader(http.StatusMethodNotAllowed)
+		}
+	}
+}
+
+// NewProxyClientFromConfig creates a new http.Client designed for proxying requests and enforcing
+// certain minimum configuration values.
+func NewProxyClientFromConfig(config types.FaaSConfig) *http.Client {
+	maxIdleConns := config.MaxIdleConns
+	if maxIdleConns < 1 {
+		maxIdleConns = 1024
+	}
+
+	maxIdleConnsPerHost := config.MaxIdleConnsPerHost
+	if maxIdleConnsPerHost < 1 {
+		maxIdleConnsPerHost = 1024
+	}
+
+	timeout := config.ReadTimeout
+	if timeout <= 0*time.Second {
+		timeout = 10 * time.Second
```

Comment:
Was this code in the gateway? It seems like the value setting should be in the code that parses / generates `types.FaaSConfig`

Reply:
I'd rather see this encapsulated in the parsing code or in a method like:

```
config.GetReadTimeout()
```

Reply:
This is in the parsing code, but if the value is the zero value, the would be a runtime bug, i figured it was better to defensively program it to make sure that it runs correctly even when set incorrectly.  

I can assume it is set correctly, but if a provider developer is manually populating that value instead of using the parser, then it could be incorrect

Reply:
What if we did a bit of both and used the value only from `config.GetReadTimeout()`? how about that?

Reply:
sounds good

Reply:
The new getters should be there now